pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -96,14 +106,30 @@ type CIDRAllocator interface {
 // CIDRAllocatorParams is parameters that's required for creating new
 // cidr range allocator.
 type CIDRAllocatorParams struct {
-	// ClusterCIDRs is list of cluster cidrs
+	// ClusterCIDRs is list of cluster cidrs.
 	ClusterCIDRs []*net.IPNet
-	// ServiceCIDR is primary service cidr for cluster
+	// ServiceCIDR is primary service cidr for cluster.
 	ServiceCIDR *net.IPNet
-	// SecondaryServiceCIDR is secondary service cidr for cluster
+	// SecondaryServiceCIDR is secondary service cidr for cluster.
 	SecondaryServiceCIDR *net.IPNet
-	// NodeCIDRMaskSizes is list of node cidr mask sizes
+	// NodeCIDRMaskSizes is list of node cidr mask sizes.
 	NodeCIDRMaskSizes []int
+	// CIDRMap maps ClusterCIDRConfig labels to ClusterCIDR objects.
+	CIDRMap map[string][]*multicidrset.ClusterCIDR
+	// AllocatedIPv4CIDRs is a map to keep track of allocated CIDRs.
+	AllocatedIPv4CIDRs map[string][]string
+	// AllocatedIPv6CIDRs is a map to keep track of allocated CIDRs.
+	AllocatedIPv6CIDRs map[string][]string
+	// CIDRLock is a mutex for CIDRMap, as it is shared with the ClusterCIDRConfig
+	// Controller.
+	CIDRLock *sync.Mutex
```

Comment:
do we need to export these variables here or can we have them inside the new controller?

Reply:
oh, do you use it to share the status between the 2 new controllers?